runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/DataflowWorkProgressUpdater.java

```diff
@@ -103,9 +112,17 @@ protected void reportProgressHelper() throws Exception {
       if (result.getHotKeyDetection() != null
           && result.getHotKeyDetection().getUserStepName() != null) {
         HotKeyDetection hotKeyDetection = result.getHotKeyDetection();
+
+        // The key set the in BatchModeExecutionContext is only set in the GroupingShuffleReader
+        // which is the correct key. The key is also translated into a Java object in the reader.
+        Object hotkey =
+            options.isHotKeyLoggingEnabled()
+                ? workItemStatusClient.getExecutionContext().getKey()
+                : null;
```

Comment:
See my earlier comment on #12984 about assuming that `key == null` is the same thing as not present.

Reply:
In the BatchModeExecutionContext the getKey method's comment states "If there is not a currently defined key, returns null." Is this an error? Do you want me to update the comment?

Reply:
That is a bug.

Reply:
Gotcha, I changed the logging